### ISSUE TITLE
Prefix Error Templates

### ISSUE THREAD

**Issue by steefaan** [cakephp/cakephp#8861]
If I create a route prefix e.g. `Admin`, I'm able to put my templates to `Templates/Admin/*`. This works fine and is pretty useful compared to Cake 2.x. My `Admin` layout is completely different to my default layout and I hoped it's also possible to create a `Templates/Admin/Error` directory. Unfortunately it isn't. I think it would be nice to add this functionality for consistency reasons, what do you think?

**Comment by markstory** [comment]
Sounds reasonable to me.

**Comment by dereuromark** [comment]
@steefaan Are you able to PR this enhancement?

**Comment by allanmcarvalho** [comment]
I solved this problem but was changing the beforeRender function inside the **ErrorController**. So, should we propose the solution to this within the **cakephp** or **app** github project?

**Comment by markstory** [comment]
Sounds like a change to be made in cakephp as that is where the default errorcontroller is for most applications.

**Comment by allanmcarvalho** [comment]
I agree @markstory , because thus updating CakePHP, everyone gets this fix. But I also had another similar problem. When using a theme, I can not create a custom layout and views within the theme, as the initializer of ErrorController overrides the initializer of the AppController which is responsible for changing the theme in my case. (This can be easily fixed in my project, but we are talking about an improvement for everyone.)
I suggest debating a solid solution to this problem, making the error layouts and views as easily customizable as all other parts of the framework, which we, we call **automagic** =D.

In summary, I suggest that by default, if the error happened on a page that uses theme and/or prefix, CakePHP first checks to see if there is a layout and view error files for that prefix or theme. If it does not exist, it loads the defaults. If it is a theme being applied over a route with a prefix, it first checks if there are files in the theme, if it does not exist, check in the prefix and if it does not exist use the default. I think that would be pretty solid. What do you think?

**Comment by markstory** [comment]
I think we should use the same theme/prefix conventions that are used elsewhere in the framework. That will lead to the least amount of surprise and minimize the documentation/testing required.

Propagating the theme is not going to be easy as themes can be set in many places during an application. But preserving the prefix sounds very doable. @allanmcarvalho Are you able to start a pull request for this change?

**Comment by dereuromark** [comment]
This also somewhat refs https://github.com/cakephp/cakephp/issues/8099 which needs the opposite to work, if no ext template then fallback to the main one.
So I guess if we can find some more dynamic/fallbacky way to do things here it would solve two problems at once.

**Comment by allanmcarvalho** [comment]
Yesterday I stayed for several hours thinking about a cool way to solve these two problems, this with the standard of quality and confibiality that makes CakePHP famous as it is. But I could not think of any. I do not know if the best solution is to modify the ErrorController or View or the default method that calls the error pages. Just think that this problem goes against all the assumptions that CakePHP says to have, that is, where it does not need a lot of configuration, everything is AutoMagic. I am willing to cooperate with this, but unfortunately I have not come to any conclusions yet.

**Comment by markstory** [comment]
Closing as a pull request is open now.
